nick_work/fdaaa_port.py

```diff
@@ -0,0 +1,372 @@
+                exported = (soup.oversight_info.is_us_export).text
+            else:
+                exported = None
+
+            #url
+            url = (soup.url).text
+
+            #official_title
+            if soup.official_title is None:
+                official_title = None
+            else:
+                official_title = (soup.official_title).text
+
+            #brief_title
+            if soup.brief_title is None:
+                brief_title = None
+            else:
+                brief_title = (soup.brief_title).txt
+
+            #title
+            if official_title is not None:
+                title = official_title
+            elif official_title is None and brief_title is not None:
+                title = brief_title
+            else:
+                title = None
+
+            #discrep_date_status - add this to "if" statement after testing: 'and completion_date is not null')
+            if (primary_completion_date < date.today() or primary_completion_date is None) and\
+            completion_date < date.today() and\
+            study_status is ("Unknown status" or "Active, not recruiting" or "Not yet recruiting" or\
+            "Enrolling by invitation" or "Suspended" or "Recruiting"):
+                discrep_date_status = 1
+            else:
+                discrep_date_status = 0
+
+            #late_cert
+            if certificate_date is not None and certificate_date > (certificate_date + relativedelta(years=1)):
+                late_cert = 1
+            else:
+                late_cert = 0
+
+            #defaulted_date
+            if (used_primary_completion_date == 1 and defaulted_pcd_flag == 1) or\
+            used_primary_completion_date == 0 and defaulted_cd_flag == 1:
+                defaulted_date = 1
+            else:
+                defaulted_date = 0
+
+            #condition
+            try:
+                condition = json.dumps(parsed_json['clinical_study']['condition'])
+            except KeyError:
+                condition = None
+
+            #condition_mesh
+            try:
+                condition_mesh = json.dumps(parsed_json['clinical_study']['condition_browse'])
+            except KeyError:
+                condition_mesh = None
+
+            #intervention
+            try:
+                intervention = json.dumps(parsed_json['clinical_study']['intervention'])
+            except KeyError:
+                intervention = None
+
+            #intervention_mesh
+            try:
+                intervention_mesh = json.dumps(parsed_json['clinical_study']['intervention_browse'])
+            except KeyError:
+                intervention_mesh = None
+
+            #keywords
+            try:
+                keywords = json.dumps(parsed_json['clinical_study']['keyword'])
+            except KeyError:
+                keywords = None
+
+            row_list = [nct_id, act_flag, included_pact_flag, has_results, pending_results, pending_data, has_certificate,\
```

Comment:
Redundant backslash

Reply:
And these are risky because you can have a value be initialized in a previous row and not set in this one, and you wouldn't catch it.

You would do better to init a dictionary at the top and think of it as your namespace, and fill it  (`v["location"] = ...`), and down here, write the dictionary. CSV module handles dictionaries. `writer.writedict(v)`.

This is also dangerous because you're assuning the same order here as in the header list above, which will change over time and eventually break. A dictionary is better.

Reply:
Thanks for the tip Chad. I might not have implemented this the way you had in mind and this might not be the best way, but weather than using `writer.writedict()` what I did instead was create a dictionary that has the eventual CSV columns as the keys, and the values are lists containing the values for each trial. I then create a header from the key values and my rows from the lists.

Let me know if there will be issues with this at scale!